### ISSUE TITLE
Typings?

### ISSUE THREAD

**Issue by cjonasw** [x2js/x2js#57]
Hey,

When compiling my Typescript I get this:

```
error TS2351: Cannot use 'new' with an expression whose type lacks a call or construct signature.
```

I thought it was missing the typings, but obviously, those come with the package, so not sure the error here.

If I ignore this the code works fine in the browser but obviously I want Typescript to understand the typings.

Any help appreciated.

**Comment by EugeneZee** [comment]
they have wrong type definitions the last string should be `export = X2JS` instead of `export default X2JS;`

**Comment by EugeneZee** [comment]
is there anyone who can merge if I make a pull request?

**Comment by sandersaares** [comment]
Sure, PRs can be merged. Perhaps @JimmyBoh, as the contributor of the typings, can comment on what is happening here.

**Comment by EugeneZee** [comment]
Here is it:
https://github.com/Axinom/x2js/pull/60

**Comment by JimmyBoh** [comment]
I originally used a default export due to this line in the documentation:

> In a Node app, require("x2js") will give you the constructor that you can use the same way.

That being said, I have no opinion either way on whether to use a default export or not. Just as long as it works!

**Comment by EugeneZee** [comment]
@JimmyBoh it doesn't work with Typescript + Webpack
Can you check that https://github.com/Axinom/x2js/pull/60 works for you?

**Comment by JimmyBoh** [comment]
Unfortunately I do not have access to the project I was working on anymore.

**Comment by cjonasw** [comment]
Works for me, who has write access within the organization ?

**Comment by cjonasw** [comment]
Maintainer needed... "This library is no longer actively used at Axinom. It therefore lacks a maintainer who can pay attention to pull requests and issues. If you are an active developer who would like to take over, please open an issue stating your intentions and providing contact information so that the maintainership can be transferred to you."

@sandersaares - @JimmyBoh and I are using this i assume in active projects so I'm good to chip in as a maintainer.

**Comment by michael-hawker** [comment]
But removing the default gives 'module x2js has no default export' with the new import syntax.

import X2JS from 'x2js';

But even then, I'm still getting 'x2js_1.default is not a constructor' from the compiled TypeScript js file output.

**Comment by EugeneZee** [comment]
you should import it like this:

import X2JS = require('x2js');

**Comment by sam-s4s** [comment]
Require is not compatible with new stuff, like rollup :(

**Comment by sandersaares** [comment]
@cjonasw sorry for the delay! I am lookig into how to best add you as a maintainer. On the whole, with this project existing under Axinom in GitHub, I would not want non-Axinom members to commit into it directly.

The ideal approach seems to be to transfer ownership to another user or organization. To make it future-proof and avoid changing URL again if a maintainer changes, I will just create a new GitHub org called x2js and add you @cjonasw as the admin for it. Makes sense? You can then give others who have showed active interest the contributor rights or organize it however you like (no objections from me no matter how you do it - as long as the package has a caring and active maintainer, I could not be happier!)

I will go ahead and do the above and then transfer this repo over to the newly created organization.

I would also like to transfer over to you the npmjs package ownership. As I understand it, this requires me to add your email address into the package metadata (and to have you registered on npmjs.com with that address). You can reveal your email address to me privately at saares@axinom.com if you prefer to avoid publishing it here in the comments.

**Comment by sandersaares** [comment]
Okay no response for a week. Still looking for maintainers. @JimmyBoh @EugeneZee are you interested? I am happy to stay active as backup admin in case of need but I do not currently have the attention to spare that the project really deserves, so if anyone wants to take over primary maintainer role, just speak up here and/or drop me an email.

**Comment by JimmyBoh** [comment]
@sandersaares Sorry I cannot be of more help, I don't have the bandwidth right now either.

**Comment by sandersaares** [comment]
I have merged #60. Please feel free to scream at me if I did a bad thing and it broke everything :) Will publish new version to npm.
